### ISSUE TITLE
undefined method `<<' for nil:NilClass

### ISSUE THREAD

**Issue by edwardloveall** [rouge-ruby/rouge#300]
When trying to lex some non-valid Apache configs, I get a strange error:

```
/Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/lexer.rb:392:in `block in lex': undefined method `<<' for nil:NilClass (NoMethodError)
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:436:in `yield'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:436:in `yield_token'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:333:in `token'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/lexers/apache.rb:43:in `block (2 levels) in <class:Apache>'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:305:in `instance_exec'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:305:in `block in step'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:287:in `each'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:287:in `step'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/regex_lexer.rb:268:in `stream_tokens'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/lexer.rb:388:in `lex'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/formatters/html.rb:50:in `each'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/formatters/html.rb:50:in `stream_untableized'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/formatters/html.rb:43:in `stream'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/formatter.rb:33:in `format'
    from /Users/edwardloveall/.gem/ruby/2.1.5/gems/rouge-1.8.0/lib/rouge/formatter.rb:25:in `format'
    from bug.rb:6:in `<main>'
```

Here's the shortest reproduction I could create:

``` ruby
require 'rouge'

formatter = Rouge::Formatter.find('html')
lexer = Rouge::Lexer.find('apache')
source = 'a;'
formatter.format(lexer.lex(source))
```

I believe this happens with other languages too, however I don't have a definitive list.

**Comment by ckruse** [comment]
I get this error as well

**Comment by pyrmont** [comment]
It looks like this issue was fixed by #385. Please let me know if that's not the case.

**Comment by ashmaroli** [comment]
Correction: This was probably fixed by #301 instead of #385 which is a bug-report ticket
